docs: Add second part of bringing dlc to lightning blog post (#47)
get10101/10101.github.io: blog/2023-07-04-bringing-dlc-to-lightning-part-2.md

---
slug: dlc-to-lightning-part-2
title: Bringing DLCs to Lightning Part 2
authors: [richard, lucas]
tags: [DLC, lightning, knowledge-base]
---

> In the first part of this blog post, we provided a comprehensive overview of Lightning, Adaptor Signatures, and DLCs.  If you require a refresher, we recommend reading [part 1](https://10101.finance/blog/dlc-to-lightning-part-1) of this series. 

Reviewer: ```suggestion
In the first part of this series, we provided an overview of Lightning, Adaptor Signatures and DLCs. If you need a refresher, you can find it [here](https://10101.finance/blog/dlc-to-lightning-part-1). 
```

![](/2023-07-04-bringing-dlc-to-lightning-part-2/bitcoin_struck_by_lightning.png)

> In part 2, we will now delve into the approach of extending the commitment transaction for a custom output. This was our first attempt on bringing DLCs to Lightning.

Reviewer: ```suggestion
In part 2, we will now delve into the approach of expanding the commitment transaction with a custom output. This was our first attempt at bringing DLCs to Lightning.
```

When coming up with the design, we knew that our modified Lightning node must still be compatible with all other regular Lightning nodes. One of the main appeals of integrating with Lightning is becoming part of the network, tapping into its fast-growing userbase.

Since we are only making additive changes, it follows that this DLC-compatible Lightning node is still able to send and receive payments with anyone else on the network.

## Lessons learned and future work

After implementing that approach as [PoC](https://github.com/get10101/10101-poc) for the [Legends of Lightning tournament](https://bolt.fun/tournaments/1/overview), we've learnt a few things which influenced how we approached our next steps with [10101](https://github.com/get10101/10101).

### Updating a channel that contains an externally-managed custom output is cumbersome

It is nice to be able to ignore the specifics of DLCs at the Lightning node level, but it makes it quite complex to update the channel state. For instance, every time a new payment is routed, there needs to be a lot of communication between the Lightning node, the layer managing the DLC and the counterparty.

### Dual-funded channels

> As mentioned, a DLC is a shared output with funds coming from two parties. As Lightning currently only supports opening unilaterally-funded channels (Certain implementations of Lightning such as Core Lightning (aka C-Lightning) do support dual-funded channels.), it is impractical to set up the channel in such a way that creating the desired DLC is possible.

Reviewer: I like the note on Core Lightning, but I would move it to a footnote. I have a tendency to add parentheses that break the flow of a sentence 🤦

## Next

> In the next and last blog post of this series we are going to write about how we are using virtual channels along the lightning channel at 10101 to bring DLCs to Lightning.

Reviewer: "In" vs "at" is probably debatable. I think we want "in" because I would refer to the project rather than the company here.

```suggestion
In the next and last blog post of this series we are going to write about how we are using virtual channels alongside the Lightning channel in 10101 to bring DLCs to Lightning.
```
